torch/distributions/categorical.py

```diff
@@ -68,6 +68,14 @@ def probs(self):
     def param_shape(self):
         return self._param.size()
 
+    @property
+    def mean(self):
+        return self.probs.new([float('nan')]).expand(self._extended_shape())
```

Comment:
That's a bit surprising. Wouldn't `(probs * arange(0, num_classes)).sum()` be a good formula for a mean here?

Reply:
That would imply ordering I guess?

Reply:
We do return ordered values. Just pointing out, I'm ok with leaving it as is